elements/radio-group/radio-group.js

```diff
@@ -1,151 +1,233 @@
+   * `RadioButton` class.
+   */
+  window.customElements.define('radio-button', RadioButton);
+
+  /**
+   * `RadioGroup` is responsible for handling user input, and updating the state
+   * of its `RadioButton` children. It uses the [roving tabindex](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Technique_1_Roving_tabindex)
+   * technique to respond to keyboard events. This ensures that the entire
+   * `RadioGroup` is a single tabstop, and tabbing into the `RadioGroup` will
+   * always focus the previously checked item, if one exists.
+   */
+  class RadioGroup extends HTMLElement {
+    constructor() {
+      super();
+    }
 
-    this.setAttribute('role', 'radiogroup');
-    // Quick and dirty set first child to tabindex 0.
-    // Really we should check to see if anyone already has tabindex 0,
-    // but we don’t care. Lol.
-    this.querySelector('[role="radio"]').setAttribute('tabindex', 0);
+    /**
+     * The `RadioGroup` sets its ARIA role to `radiogroup` and sets the
+     * `tabindex` on its first `RadioButton` child to 0 if no other child is
+     * already set to `tabindex=0`. This makes the first `RadioButton` focusable.
+     * This assumes `RadioButton` children are already in the DOM and their
+     * definition has already been loaded. For a more robust implementation you
+     * might consider using a Mutation Observer to detect if children are
+     * present yet. The `RadioGroup also adds listeners for keyboard and mouse
+     * events. 
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'radiogroup');
+      let hasCheckedChild = this.querySelector('[role="radio"][tabindex="0"]');
+      if (!hasCheckedChild) {
+        this.querySelector('[role="radio"]').setAttribute('tabindex', 0); 
+      }
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
 
-  }
+    /**
+     * If the `RadioGroup` is removed from the DOM, clean up any event
+     * listeners.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('click', this._onClick);
+    }
 
-  onKeyDown(e) {
-    switch (e.keyCode) {
-      case KEYCODE.UP:
-      case KEYCODE.LEFT:
-        {
-          // We will be doing all the handling, so we don’t want
-          // the browser to do anything for us.
+    /**
+     * If the user pressed an arrow key, call preventDefault to prevent the
+     * page from scrolling. If the up or left arrow keys were pressed, select
+     * the previous `RadioButton`. If the down or right keys were pressed,
+     * select the next `RadioButton`.
+     */
+    _onKeyDown(e) {
+      switch (e.keyCode) {
+        case KEYCODE.UP:
+        case KEYCODE.LEFT:
           e.preventDefault();
-          _this.setCheckedToPrevButton();
+          this._setCheckedToPrevButton();
```

Comment:
This is one of my favorite parts of custom elements. Because the DOM element is an actual instance of this class, `this` will refer to the class object. Way better than passing DOM elements into a constructor to wrap them.